### ISSUE TITLE
ntvdm.exe spins the processor on 100% and does not start the program

### ISSUE THREAD

**Issue by CosworthF1** [leecher1337/ntvdmx64#27]
I want to run an old 16-bit Windows application on Windows 7 and Windows 10 64-bit operating systems. The following error occurs: the program you want to run does not start, and ntvdm.exe loads the processor on 100% and runs until it shoots. I downloaded the program from http://www.columbia.edu/~em36/ntvdmx64.html and installed it, apparently without error. What are you suggestin what to look at?
I've tried several computers, none of them started. I've shut down the virus scanner and firewall so that this is not the problem.  Windows 7 and Windows 10 on 32-bit systems work well.

DebugView log:

00000001	0.00000000	[6548] LDNTVDM: BasepProcessInvalidImage(C0000130,'\??\C:\ardesign\ardesign.exe');	
00000002	0.00014405	[6548] LDNTVDM: Launching Win16 application	
00000003	0.00295506	[6548] LDNTVDM: BasepProcessInvalidImage = 1	
00000004	0.05852314	[6820] Hook_Inline(775477A0, F29C2284, 0)	
00000005	0.05859095	[6820] dwOrigSize detected: 10	
00000006	0.05865289	[6820] Hook_Inline context=7754BF29	
00000007	0.05870531	[6820] Hook_IAT_x64_IAT(774B0000, ntdll.dll, NtCreateUserProcess, F29C20E8, F29CB8B8)	
00000008	0.06109076	[6820] LDNTVDM: BasepProcessInvalidImageReal = 7754BF29	
00000009	0.06114831	[6820] LDNTVDM: BaseIsDosApplication = 774F9680	
00000010	0.06120732	[6820] LDNTVDM is running inside ConHost.exe	
00000011	0.06132498	[6820] ConsBmp_Install done	
00000012	0.06136823	[6820] Hook_IAT_x64_IAT(FFB00000, ntdll.dll, RtlAllocateHeap, F29C27E0, F29CB8B0)	
00000013	0.06144337	[6820] Hook_IAT_x64_IAT(FFB00000, user32.dll, NotifyWinEvent, F29C2864, F29CB8A8)	
00000014	0.07851432	[6856] RegQueryValueEx 1 failed: 00000002	
00000015	0.07858177	[6856] RegQueryValueEx 2 failed: 00000002	
00000016	0.07863638	[6856] Hook_IAT_x64_IAT(76D20000, ntdll.dll, NtCreateUserProcess, 711126B0, 7111AE28)	
00000017	0.07939878	[6856] LDNTVDM: BasepProcessInvalidImageReal = 00000000	
00000018	0.07944863	[6856] LDNTVDM: BaseIsDosApplication = 00000000	
00000019	0.07951277	[6856] Hook_IAT_x64_IAT(76D20000, ntdll.dll, CsrClientCallServer, 71111FB0, 71118008)	
00000020	0.07964143	[6856] Hook_IAT_x64_IAT(76D20000, ntdll.dll, CsrAllocateMessagePointer, 711121B0, 00000000)	
00000021	0.07977302	[6856] Hook_IAT_x64_IAT(0F000000, KERNEL32.DLL, SetConsolePalette, 71112D60, 7111AE24)	
00000022	0.07992771	[6856] Hook_IAT_x64_IAT(76D20000, ntdll.dll, NtQueryInformationProcess, 71112C20, 00000000)	
00000023	0.08019271	[6856] Hook_IAT_x64_IAT(76E30000, ntdll.dll, NtQueryInformationProcess, 71112C20, 00000000)	
00000024	0.08140706	[6856] BaseGetNextVDMCommand: ConsoleHandle=00001AA4, iTask=00000016	
00000025	0.08149944	[6856] BaseGetNextVDMCommand(268500999) = C000000D	
00000026	0.08170470	[6856] NTVDM: 25600K Memory: 8192K XMS, 0K EMS, 16384K DPMI	
00000027	0.11795566	[6856] BaseIsFirstVDM(65545) = C0000022	
00000028	14.51950359	[7004] RegQueryValueEx 1 failed: 00000002	
00000029	15.01868629	[7004] Hook_IAT_x64_IAT(76D20000, ntdll.dll, NtCreateUserProcess, 711126B0, 7111AE28)	
00000030	15.53843021	[7004] LDNTVDM: BaseIsDosApplication = 00000000	
00000031	16.03252602	[7004] Hook_IAT_x64_IAT(76D20000, ntdll.dll, CsrAllocateMessagePointer, 711121B0, 00000000)	
00000032	16.54724693	[7004] Hooking failed.	
00000033	16.54757881	[7004] Hook_IAT_x64_IAT(76D20000, ntdll.dll, NtQueryInformationProcess, 71112C20, 00000000)	
00000034	16.54856682	[7004] Hook_IAT_x64_IAT(76E30000, ntdll.dll, NtQueryInformationProcess, 71112C20, 00000000)

**Comment by emendelson** [comment]
As it says on the page that you linked to, NTVDMX64 makes it possible to run DOS applications. It does nothing for 16-bit Windows applications. You might want to try WIN3MU (search for it - it's open source), but WIN3MU doesn't support a wide range of applications. Otherwise, you'll need to use DOSBox or something similar running Windows 3.1.

**Comment by CosworthF1** [comment]
Thank you very much for your answer!

My problem was the following: I had to have a solution that was a 32-bit application launching a 16-bit application, so unfortunately the Dosbox solution would not be good to me because it would have been a bit macerated, it would have burdened the resources of the computer. Win3mu would be good, but I would have to translate it, which unfortunately I do not understand because the exe setup file is not available on the internet... But I've been browsing the web and I've found the right solution for me: the WineVDM project is, what I was looking for and the program I started to run with the developer version :) I'll write a link to the website if it's useful for you (or for somebody else) in the near future:

https://github.com/otya128/winevdm

(If you're down on the page below, you can download the finished translated files)

**Comment by leecher1337** [comment]
Very cool finding, thank you!

I did a port of all the necessary 16->32bit APIs and such stuff, so the translation backend in NTVDMx64 should work fine, but unfortunately, functionality to sync the task with the Windows NT task scheduler and Win32k subsystem is missing on x64 Windows leading to random crashes and inproper execution on the CCPU, which finally makes it impossible to use it in NTVDMx64. I ran out of ideas how to compensate the missing functions of the Windows subsystem on x64, therefore I never released my patches.
If someone is interested in the patches for the translation layer of WOW16 (together with leaked Windows 2000 sources and a lot of manual work, I was able to reconstruct Win2k+ WOW16 layer, which is a bit different to NT4), I can upload them, but as there are currently mainly users for this project and not many developers with in-depth Windows internals knowledge, I doubt it is worth the hassle of creating patch-files, or is it...?
